### DISCUSSION TITLE
Non-expire option for Granular Access Token

### DISCUSSION THREAD

**Discussion by felixle236** [npm/feedback#849] [Registry] [answered]
I have more than 20 private packages and I want to manage the permission for access tokens. So "Granular Access Token" is very useful for us, but we can set the expiration time from 1-356 days, I want to suggest another option as "non-expire", I think it's very important and suitable for this feature.

Thanks

**Comment by actuallymentor** [comment]
I'd like to stress here that for usecases like this the current setup allows these two options:

1. A token that has eternal access to write everything
2. A granular token with access to a single package, but CI pipelines will break every year because the token expired

I'm not sure how the max expiry decision was made but this seems like forcing the developer to choose between bad security or bad operations management.

**Comment by monishcm** [comment, marked as answer]
Thanks for the feedback. We are working on new features to overcome this issue.
